app/src/main/java/com/example/bigowlapp/utils/AlarmBroadcastReceiverManager.java

```diff
@@ -0,0 +1,120 @@
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.repository.ScheduleRepository;
+import com.example.bigowlapp.service.AlarmBroadcastReceiver;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.Timestamp;
+import com.google.firebase.firestore.GeoPoint;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static com.example.bigowlapp.utils.IntentConstants.EXTRA_LATITUDE;
+import static com.example.bigowlapp.utils.IntentConstants.EXTRA_LONGITUDE;
+import static com.example.bigowlapp.utils.IntentConstants.EXTRA_UID;
+
+/**
+ * The purpose of this class is to set/define alarms that the app will set.
+ * After the time activation of an alarm, code will be executed from
+ * {@link com.example.bigowlapp.service.AlarmBroadcastReceiver}
+ */
+public class AlarmBroadcastReceiverManager {
+
+    private final Context context;
+    private final ScheduleRepository scheduleRepository;
+
+    public AlarmBroadcastReceiverManager(Context context) {
+        this.context = context;
+        scheduleRepository = new ScheduleRepository();
+    }
+
+    // TODO: --START-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+    public AlarmBroadcastReceiverManager(Context context, ScheduleRepository scheduleRepository) {
+        this.context = context;
+        this.scheduleRepository = scheduleRepository;
+    }
+    // TODO: --END-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+
+    /**
+     * Sets the alarm(s) for the BroadcastReceiver given the schedules that the user has
+     *
+     * @param userID The Id of the user
+     */
+    public void setAlarms(String userID) {
+        int requestCode = 0;
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        getSchedulesForUser(userID)
+                .addOnSuccessListener(scheduleList -> {
+                    for (Schedule schedule : scheduleList) {
+                        Intent intent = new Intent(context, AlarmBroadcastReceiver.class);
+                        intent.putExtra(EXTRA_UID, schedule.getUid());
+                        intent.putExtra(EXTRA_LATITUDE, schedule.getLocation().getLatitude());
+                        intent.putExtra(EXTRA_LONGITUDE, schedule.getLocation().getLongitude());
+                        PendingIntent pendingIntent = PendingIntent
+                                .getBroadcast(context, requestCode, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                        alarmManager.set(AlarmManager.RTC_WAKEUP,
+                                schedule.getStartTime().toDate().getTime(), pendingIntent);
+                    }
+                }).addOnFailureListener(error -> Toast.makeText(context, error.getMessage(), Toast.LENGTH_SHORT).show());
```

Comment:
What type of error would show here btw. If it would be confusing to the user, might be better to simply Log the error, but depends what a failure results in.

Reply:
I would display an error concerning querying schedules.

In practice, this error shouldn't be reached, but, for some exceptional cases that Firestore throws an error, the error that is thrown by Firestore will be displayed to the user. I believe that Firestore would throw errors with useful message.